crates/compiler/src/obj_gen/mod.rs

```diff
@@ -553,6 +553,302 @@ impl ObjectGenerator {
+        // output _value_ and an input block reference.
+        let mut incoming_value_ids = Vec::new();
+        let mut incoming_value_names = Vec::new();
+        let mut incoming_value_types = Vec::new();
+        let mut incoming_block_ids = Vec::new();
+
+        for (value, block) in phi_instruction.get_incomings() {
+            // We start by resolving the incoming value to an identifier, handling both
+            // constants and variables.
+            incoming_value_names.push(value.get_name().to_str()?.to_string());
+            incoming_value_ids.push(get_var_or_const(&value, bb, func_ctx)?);
+            incoming_value_types.push(LLVMType::try_from(value.get_type())?);
+
+            // We then need to check that we know of the block, and its identifier.
+            incoming_block_ids.push(func_ctx.try_lookup_block(block.get_name().to_str()?)?);
+        }
+
+        // Phi is represented by a function, so we need a signature. It has to take both
+        // the "source block variable", and the placeholders for the variables used
+        // _within_ the function. The type consists of the variable to be filled with
+        // the block id, followed by all the variables that need to be passed in, in
+        // the same order as they appear in the phi instruction.
+        let mut func_params = incoming_value_types.iter().cloned().collect_vec();
+        func_params.insert(0, LLVMType::i128);
+
+        // The signature of the phi needs to take variables that are fresh in the
+        // _context of the new phi function_.
+        let mut phi_func_params = func_params
+            .iter()
+            .map(|typ| Ok(bb.add_variable(ObjectContext::flo_type_of(typ)?)))
+            .collect::<Result<Vec<_>>>()?;
+        let incoming_var = bb.add_variable(Type::Unsigned128);
+        phi_func_params.insert(0, incoming_var);
+
+        let phi_sig = Signature {
+            params:   phi_func_params.clone(),
+            returns:  vec![bb.add_variable(phi_result_type_flo.clone())],
+            location: None,
+        };
+
+        // We need to create a function that implements the phi. For this, we take the
+        // block condition and turn it into a match check. We then iterate over each
+        // incoming variable, and pick the one corresponding to the block in question.
+        let phi_func = bb.context.add_function(&phi_sig, |bb| -> Result<()> {
+            // We start by getting the first of the parameters, which is the one we need to
+            // compare against. It is safe to index directly as we know it must exist here.
+            let compare_param = &phi_func_params[0];
+
+            // We start by creating the condition variable for each. This variable is either
+            // true or false, and will only be true if the incoming block variable matches
+            // the block identifier for that match arm.
+            let condition_variables = phi_func_params
```

Comment:
Honestly, I feel like blocks like this are _less_ clear functionally and more clear with a simpler for, especially where you can do things like comment why you're `continue`'ing on the first variable.,